services/rule_engine/src/main/java/agents/Smith.java

```diff
@@ -1,27 +1,77 @@
 package agents;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
+import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI;
 import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet;
 import com.mindsmiths.ruleEngine.model.Agent;
+import com.mindsmiths.ruleEngine.util.Agents;
 import com.mindsmiths.ruleEngine.util.Log;
 
+
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.ToString;
 
+
 @Data
 @ToString(callSuper = true)
 @AllArgsConstructor
 public class Smith extends Agent {
 
     LocalDateTime lastTableUpdate = null;
+    int nextRow = 0;
+    int odgNext = 0;
+    Spreadsheet sheet;
 
+    // Agent id
     public Smith() {
         this.id = "SMITH";
     }
 
+    // Getting the sheet as an object and getting the first free row in the spreadsheet
     public void processSheet(Spreadsheet spreadsheet){
```

Comment:
You're processing a spreadsheet, but the function is called `processSheet`

Reply:
done